pkg/standalone-utils/contracts/base/MultiDelegatecall.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Address.sol";
+
+abstract contract MultiDelegatecall {
+    using Address for address payable;
+
+    function multicall(bytes[] calldata data) external payable returns (bytes[] memory results) {
+        results = new bytes[](data.length);
+        for (uint256 i = 0; i < data.length; i++) {
+            // solhint-disable-next-line avoid-low-level-calls
+            (bool success, bytes memory result) = address(this).delegatecall(data[i]);
+
+            if (!success) {
+                // Next 6 lines from https://ethereum.stackexchange.com/a/83577
+                if (result.length < 68) revert("MULTICALL_FAILED");
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    result := add(result, 0x04)
+                }
+                revert(abi.decode(result, (string)));
```

Comment:
```suggestion
                assembly {
                    returndatacopy(0, 0, returndatasize())
                    revert(0, returndatasize())
                }
```

We should have tests for reverts that happen both in the multicall (the sender `require`s) and in the Vault to test the the revert reason bubble-up mechanism works correctly.

Reply:
Addressed in #671